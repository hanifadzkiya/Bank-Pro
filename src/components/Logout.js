import React, { Component } from "react";

class Logout extends Component {
    constructor(props){
        super(props);
        this.logout = this.logout.bind(this);
    }

    logout(){
        sessionStorage.clear();
    }

    render() {
        return (
            <div>
                <h2>Logout</h2>
                <p>Tekan tombol untuk logout</p>
                <button onClick={this.logout}>Logout</button>
            </div>
        );
    }
}

export default Logout;
