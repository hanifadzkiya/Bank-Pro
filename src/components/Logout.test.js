import React from "react";
import Logout from './Logout';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';

it('Render Logout', () => {
    const wrapper = shallow(<Logout />);
    expect(wrapper.contains(<h2>Logout</h2>)).toEqual(true);
    expect(wrapper.contains(<p>Tekan tombol untuk logout</p>)).toEqual(true);

    expect(wrapper).toMatchSnapshot();
})