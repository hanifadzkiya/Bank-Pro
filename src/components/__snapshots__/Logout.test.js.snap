// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Logout 1`] = `ShallowWrapper {}`;
